sql/src/test/java/org/opensearch/sql/sql/antlr/SQLSyntaxParserTest.java

```diff
@@ -191,6 +191,12 @@ public void can_parse_now_like_functions(String name, Boolean hasFsp, Boolean ha
     assertNotNull(parser.parse("SELECT id FROM test WHERE " + String.join(" AND ", calls)));
   }
 
+  @Test
+  public void can_parse_month_of_year_function() {
+    assertNotNull(parser.parse("SELECT month('2022-11-18')"));
+    assertNotNull(parser.parse("SELECT month_of_year('2022-11-18')"));
```

Comment:
What happens with invalid dates (eg. 13th month, 31st day in February etc).

Reply:
At this stage I believe the tests should pass. Because it is still a valid format as far as the lexer/parser is concerned.